examples/private_set_intersection/client/rust/src/main.rs

```diff
@@ -0,0 +1,187 @@
+use log::info;
+use maplit::hashset;
+use oak_abi::label::{web_assembly_module_signature_tag, Label};
+use private_set_intersection_client::proto::{
+    private_set_intersection_client::PrivateSetIntersectionClient, SubmitSetRequest,
+};
+use prost::Message;
+use std::collections::HashSet;
+use structopt::StructOpt;
+use tonic::{
+    metadata::MetadataValue,
+    transport::{Certificate, Channel, ClientTlsConfig},
+    Request,
+};
+
+#[derive(StructOpt, Clone)]
+#[structopt(about = "Private Set Intersection Client")]
+pub struct Opt {
+    #[structopt(
+        long,
+        help = "URI of the Oak application to connect to",
+        default_value = "https://localhost:8080"
+    )]
+    uri: String,
+    #[structopt(
+        long,
+        help = "PEM encoded X.509 TLS root certificate file used by gRPC client"
+    )]
+    root_tls_certificate: String,
+    #[structopt(long, help = "Path to the PEM-encoded public key used as a data label")]
+    public_key: String,
+}
+
+/// Create a TLS channel for connecting to Oak.
+async fn create_channel(uri: &Uri, root_tls_certificate: &[u8]) -> anyhow::Result<Channel> {
+    info!("Connecting to Oak Application: {:?}", uri);
+    let tls_config =
+        ClientTlsConfig::new().ca_certificate(Certificate::from_pem(root_tls_certificate));
+    Channel::builder(uri.clone())
+        .tls_config(tls_config)
+        .context("Couldn't create TLS configuration")?
+        .connect()
+        .await
+        .context("Couldn't connect to Oak Application")
+}
+
+/// Create Oak gRPC client.
+async fn create_client(
+    channel: Channel,
+    public_key: &[u8],
+) -> anyhow::Result<PrivateSetIntersectionClient<Channel>> {
+    let public_key_tag = web_assembly_module_signature_tag(&public_key);
+    let mut label = Vec::new();
+    Label {
+        confidentiality_tags: vec![public_key_tag],
+        integrity_tags: vec![],
+    }
+    .encode(&mut label)
+    .context("Error encoding label")?;
+    Ok(PrivateSetIntersectionClient::with_interceptor(
+        channel,
+        move |mut request: Request<()>| {
+            request.metadata_mut().insert_bin(
+                oak_abi::OAK_LABEL_GRPC_METADATA_KEY,
+                MetadataValue::from_bytes(label.as_ref()),
+            );
+            Ok(request)
+        },
+    ))
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    env_logger::init();
+    let opt = Opt::from_args();
+
+    let uri = opt.uri.parse().context("Error parsing URI")?;
+    let root_tls_certificate = tokio::fs::read(&opt.root_tls_certificate)
+        .await
+        .context("Couldn't load certificate file")?;
+    let public_key_file = tokio::fs::read(&opt.public_key)
+        .await
+        .context("Couldn't load public key file")?;
+    let public_key_bytes = pem::parse(public_key_file)
+        .context("Empty public key file")?
+        .contents;
+
+    // Submit sets from different clients.
+    let channel_1 = create_channel(&uri, &root_tls_certificate)
```

Comment:
Who does 1-based indexing? :)

Reply:
We do =)
https://github.com/project-oak/oak/blob/754c3c9d49939f8fa177bef80b7a0e8a99be1f5c/examples/private_set_intersection/client/private_set_intersection.cc#L72-L78

Reply:
Not sure if it's better to rename it to `first_channel` etc.
Since we need to connect to Oak from two distinct clients.

Reply:
The c++ code you linked does 0-based indexing, right? (Not that it matters much, but not sure I missed your point)

first and second seem worse options to me, we already have numbers that we can use to enumerate things, why use words? :)